packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -39,90 +86,199 @@ const ConfirmSignupCode = ({
+        visible: true,
+        type: BannerType.error,
+        message: localizedBannerMessage,
+      });
     }
+  }
+
+  function alertSuccessAndGoForward() {
+    alertBar.success(
+      ftlMsgResolver.getMsg(
+        'confirm-signup-code-success-alert',
+        'Account confirmed successfully'
+      )
+    );
+    // TODO redirect elsewhere if relying party
+    navigate('/settings', { replace: true });
+  }
+
+  async function verifySession(code: string) {
+    logViewEvent(`flow.${viewName}`, 'submit', REACT_ENTRYPOINT);
     try {
-      // Check confirmation code
-      // Log success event (verification.success)
-      // Display success message
-      // Handle newsletter subscription
+      await account.verifySession(code);
+
+      // FOLLOW-UP: does not yet exist in Settings
+      // BEFORE: notifier.trigger('verification.success');
+      logViewEvent(
+        `flow.${viewName}`,
+        'verification.success',
+        REACT_ENTRYPOINT
+      );
+      // FOLLOW-UP: enable once state can be accessed from previous component
+      // FOLLOW-UP: notifier does not exist in Settings
+      // BEFORE: this.notifier.trigger('flow.event', {event: 'newsletter.subscribed',})
+      // if (newsletters) {
+      //   logViewEvent(`flow.${viewName}`, 'newsletter.subscribed', REACT_ENTRYPOINT);
+      // }
+
+      // FOLLOW-UP: Broker not yet implemented
+      // await broker.invokeBrokerMethod('afterSignUpConfirmationPoll', account);
+      alertSuccessAndGoForward();
     } catch (e) {
-      // TODO: error handling, error message confirmation
-      // Possible AuthErrors to display in tooltip:
-      //   - 'INVALID_EXPIRED_SIGNUP_CODE' = 'Invalid or expired confirmation code'
-      //   - 'OTP_CODE_REQUIRED' = 'Please enter confirmation code'
-      //   - 'INVALID_OTP_CODE' = 'Valid code required'
-      // all other errors: alertBar.error(errorConfirmSignupCode);
+      // TODO log error
+      const localizedErrorMessage = ftlMsgResolver.getMsg(
+        composeAuthUiErrorTranslationId(e),
+        e.message
+      );
+      // If the error is one of the three indicated types, display error in input tooltip
+      if (
+        e.errno ===
+        (AuthUiErrors.INVALID_EXPIRED_SIGNUP_CODE.errno ||
+          AuthUiErrors.OTP_CODE_REQUIRED.errno ||
+          AuthUiErrors.INVALID_OTP_CODE.errno)
+      ) {
+        setBanner({ visible: false, type: undefined, message: undefined });
+        setCodeErrorMessage(localizedErrorMessage);
+      } else {
+        // Any other error messages should be displayed in an error banner
+        // Should we display a generic user-friendly error?
+        // Right now this shows errors like "Invalid parameter in request body"
+        setBanner({
+          visible: true,
+          type: BannerType.error,
+          message: localizedErrorMessage,
+        });
+      }
     }
-  };
+  }
+
+  const localizedPageTitle = ftlMsgResolver.getMsg(
+    'confirm-psignup-code-page-title',
```

Comment:
Should this be "confirm-signup-code-page-title" (possibly typo'd `p` in `psignup`).